python/lsst/daf/butler/datastores/fileLikeDatastore.py

```diff
@@ -825,7 +830,69 @@ def _write_in_memory_to_artifact(self, inMemoryDataset: Any, ref: DatasetRef) ->
         info : `StoredFileInfo`
             Information describin the artifact written to the datastore.
         """
-        raise NotImplementedError()
+        location, formatter = self._prepare_for_put(inMemoryDataset, ref)
+        uri = location.uri
+
+        if uri.exists():
+            # Assume that by this point if registry thinks the file should
+            # not exist then the file should not exist and therefore we can
+            # overwrite it. This can happen if a put was interrupted by
+            # an external interrupt. The only time this could be problematic is
+            # if the file template is incomplete and multiple dataset refs
+            # result in identical filenames.
+            # Eventually we should remove the check completely (it takes
+            # non-zero time for network).
+            log.warning("Object %s exists in datastore for ref %s", uri, ref)
```

Comment:
Is it time to just remove this check now?

Reply:
Ok.